### DISCUSSION TITLE
UI intergration

### DISCUSSION THREAD

**Discussion by treezy254** [microsoft/promptflow#1946] [Q&A] [answered]
I've been developing a system using Promptflow for quite a while now. What I would like to inquire about is whether if it's possible to integrate a prompt flow with an external or any library that provides UI functionalities such as Chainlit

**Comment by brynn-code** [comment, marked as answer]
Hi @treezy254. thanks for reaching out, we have streamlit lib integrate with some of our command, by installing promptflow[executable] and running `pf flow test --flow <flow-path> --ui`, a web browser window will be pop up, and the UI is provided by streamlit. We will generate the source file under the temp folder, I'm sorry that we don't log the source path out (we will add the log message), you could find them under the temp folder, looks like: 
![image](https://github.com/microsoft/promptflow/assets/24237253/1aff6096-1405-4b02-bb22-3eb3ca165a75)
And remember to copy them out, because once you stop the streamlit server, the files will be removed.
